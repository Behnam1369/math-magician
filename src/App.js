import React from 'react';
import Calculator from './components/Calculator';

class App extends React.PureComponent {
  render() {
    return (
      <div>
        <Calculator />
      </div>
    );
  }
}

export default App;
