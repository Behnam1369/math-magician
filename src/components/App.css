body {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  background-color: rgb(29, 1, 29);
  color: rgb(255, 0, 234);
  margin: 0;
}

header {
  display: flex;
  justify-content: space-around;
  background-color: #190019;
  margin-bottom: 20px;
  border-bottom: solid 1px rgba(192, 192, 192, 0.301);
}

header h1 {
  color: yellow;
  display: inline;
}

header nav {
  display: inline;
}

header ul {
  display: inline-flex;
}

header li {
  list-style: none;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  padding: 10px;
}

header li a {
  color: yellow;
  transition: all 0.2s;
  cursor: pointer;
  text-decoration: none;
}

header li a:hover {
  color: white;
  border-bottom: dotted 1px white;
}

main {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

main h2,
main p {
  width: 90%;
  max-width: 1000px;
}

.calcContainer {
  display: flex;
  justify-content: space-around;
}

.calcContainer > div {
  text-align: center;
}

.display {
  border: solid 1px white;
}
