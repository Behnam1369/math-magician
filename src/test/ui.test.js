/* eslint-disable no-unused-vars */
import React from 'react';
import {
  render, fireEvent, screen, waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../App';
import operate from '../logic/operate';
import calculate from '../logic/calculate';

// Set up React Testing Library.
// Follow the instructions in the official documentation.
// You don't need to setup Jest (it's included in Create React App).
// Write unit tests for the files operate.js and calculate.js using Jest.
// Create unit tests for all React components:
// Use Jest snapshots to test the components.
// Use React Testing Library to simulate user interaction.
// Run your full tests suite using Jest. All tests should pass.

describe('operate(numberOne, numberTwo, operation)', () => {
  it('should add numbers', () => {
    expect(operate('2', '4', '+')).toBe('6');
    expect(operate('4', '4', '+')).toBe('8');
    expect(operate('5', '4', '+')).toBe('9');
  });
  it('should substract numbers', () => {
    expect(operate('2', '4', '-')).toBe('-2');
    expect(operate('4', '4', '-')).toBe('0');
    expect(operate('5', '4', '-')).toBe('1');
  });
  it('should multiply numbers', () => {
    expect(operate('2', '4', 'x')).toBe('8');
    expect(operate('4', '4', 'x')).toBe('16');
    expect(operate('5', '4', 'x')).toBe('20');
  });
  it('should divide numbers', () => {
    expect(operate('2', '4', '÷')).toBe('0.5');
    expect(operate('4', '4', '÷')).toBe('1');
    expect(operate('5', '4', '÷')).toBe('1.25');
    expect(operate('5', '0', '÷')).toBe('Can\'t divide by 0.');
  });
  it('should calculate division module', () => {
    expect(operate('2', '4', '%')).toBe('2');
    expect(operate('4', '4', '%')).toBe('0');
    expect(operate('5', '4', '%')).toBe('1');
    expect(operate('5', '0', '%')).toBe('Can\'t find modulo as can\'t divide by 0.');
  });
});

describe('calculate(obj, buttonName)', () => {
  it('should react to numbers', () => {
    expect(calculate({ total: null, next: null, operation: null }, '4')).toEqual({ total: null, next: '4', operation: null });
    expect(calculate({ total: null, next: '3', operation: null }, '5')).toEqual({ total: null, next: '35', operation: null });
    expect(calculate({ total: '4', next: null, operation: '+' }, '4')).toEqual({ total: '4', next: '4', operation: '+' });
  });
  it('should react to operatios', () => {
    expect(calculate({ total: null, next: '34', operation: null }, '+')).toEqual({ total: '34', next: null, operation: '+' });
    expect(calculate({ total: null, next: '34', operation: null }, '-')).toEqual({ total: '34', next: null, operation: '-' });
    expect(calculate({ total: null, next: '34', operation: null }, 'x')).toEqual({ total: '34', next: null, operation: 'x' });
    expect(calculate({ total: null, next: '34', operation: null }, '%')).toEqual({ total: '34', next: null, operation: '%' });
    expect(calculate({ total: null, next: '34', operation: null }, '÷')).toEqual({ total: '34', next: null, operation: '÷' });
    expect(calculate({ total: '9', next: '3', operation: '+' }, '-')).toEqual({ total: '12', next: null, operation: '-' });
  });
  it('should multiply numbers', () => {
    expect(calculate({ total: null, next: null, operation: null }, '4')).toEqual({ total: null, next: '4', operation: null });
    expect(calculate({ total: null, next: '3', operation: null }, '5')).toEqual({ total: null, next: '35', operation: null });
    expect(calculate({ total: '4', next: null, operation: '+' }, '4')).toEqual({ total: '4', next: '4', operation: '+' });
  });
  it('should divide numbers', () => {
    expect(operate('2', '4', '÷')).toBe('0.5');
    expect(operate('4', '4', '÷')).toBe('1');
    expect(operate('5', '4', '÷')).toBe('1.25');
    expect(operate('5', '0', '÷')).toBe('Can\'t divide by 0.');
  });
  it('should calculate division module', () => {
    expect(operate('2', '4', '%')).toBe('2');
    expect(operate('4', '4', '%')).toBe('0');
    expect(operate('5', '4', '%')).toBe('1');
    expect(operate('5', '0', '%')).toBe('Can\'t find modulo as can\'t divide by 0.');
  });
});

test('two minus eight should return -6. ', async () => {
  render(<App />);

  fireEvent.click(screen.getByText('Calculator'));
  await waitFor(() => screen.getByText('4'));
  fireEvent.click(screen.getByText('2'));
  fireEvent.click(screen.getByText('-'));
  fireEvent.click(screen.getByText('8'));
  fireEvent.click(screen.getByText('='));

  expect(screen.getByTestId('display')).toHaveValue('-6');
});
