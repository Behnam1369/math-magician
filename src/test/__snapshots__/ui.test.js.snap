// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing react components using jest snapshot Testing App component snapshot 1`] = `
<div>
  <header>
    <h1>
      Math Magician
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="#/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#/Calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="#/Quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </main>
</div>
`;

exports[`testing react components using jest snapshot Testing Button component snapshot 1`] = `
<button
  className="silver"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;

exports[`testing react components using jest snapshot Testing Calculator component snapshot 1`] = `
<div
  className="calcContainer"
>
  <div>
    <h2>
      Let's do some math
       
      <br />
       
      Feel free using your keyboard
    </h2>
  </div>
  <div
    className="calculator"
  >
    <input
      className="display"
      data-testid="display"
      disabled={true}
      type="text"
      value=""
    />
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="silver btn0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="silver"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`testing react components using jest snapshot Testing Header component snapshot 1`] = `
<header>
  <h1>
    Math Magician
  </h1>
  <nav>
    <ul>
      <li>
        <a
          href="#/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="#/Calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="#/Quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`testing react components using jest snapshot Testing Home component snapshot 1`] = `
<main>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</main>
`;

exports[`testing react components using jest snapshot Testing Quote component snapshot 1`] = `
<h2
  style={
    Object {
      "margin": "auto",
      "maxWidth": "1000px",
      "textAlign": "center",
    }
  }
>
  Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding. - William Paul Thurston
</h2>
`;
